lib/chitter.rb

```diff
@@ -0,0 +1,83 @@
+require 'sinatra/base'
+require 'data_mapper'
+require 'dm-migrations'
+require 'bcrypt'
+require './lib/user'
+require './lib/peep'
+
+
+class Chitter < Sinatra::Base
+  set :views, proc {File.join(root, '..', 'views')}
+  enable :sessions
+  set :session_secret, '123321123'
+  use Rack::Session::Pool
+  env = ENV['RACK_ENV'] || "development"
+
+  DataMapper.setup(:default, ENV['DATABASE_URL'] || "postgres://localhost/chitter_#{env}")
+  DataMapper.finalize
+  DataMapper.auto_upgrade!
+  #DataMapper.auto_migrate!
+  DataMapper::Model.raise_on_save_failure = true
+
+  before do
+    @user = User.get(session[:user_id]) unless is_user?
+  end
+
+  def is_user?
+    @user != nil
+  end
+
+  get '/' do
+    @peeps = Peep.all
+    erb :index
+  end
+
+  get '/sign_up' do
+    erb :sign_up
+  end
+
+  post '/sign_up' do
+    begin
+      @user = User.create(:username => params[:username], :password => params[:password], :password_confirmation => params[:password_confirmation], :name => params[:name], :email => params[:email],)
```

Comment:
Metrics/LineLength: Line is too long. [198/90]<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax.